src/RatcliffDDM.jl

```diff
@@ -1,371 +1,605 @@
-
-#     Numerical Integration with Simpson's Method
-#     see: https://en.wikipedia.org/wiki/Simpson%27s_rule
-
-# # Arguments
-# - `x`::Real: response time
-# - `ν`::Real: response time
-# - `η`::Real: response time
-# - `α`::Real: response time
-# - `z`::Real: response time
-# - `τ`::Real: response time
-# - `ϵ`::Real: response time
-# - `lb_z`::Real: response time
-# - `ub_z`::Real: response time
-# - `n_sz`::Int: response time
-# - `lb_t`::Real: response time
-# - `ub_t`::Real: response time
-# - `n_st`::Int: response time
-
-# """
-# function _simpson_1D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
-#         #assert ((n_sz&1)==0 and (n_st&1)==0), "n_st and n_sz have to be even"
-
-# end
-
-# function _simpson_2D()
-    
-# end
-
-# """
-# cdf_full()
-
-#  The orignial algorithm was written on 09/01/06 by Joachim Vandekerckhove
-#  Then converted from c to julia by Kianté Fernandez
- 
-#  Computes Cumulative Distribution Function for the Diffusion model with random trial to trial mean drift (normal), 
-#  starting point and non-decision (ter) time (both rectangular). Uses 6 quadrature points for drift and 6 for the others.
-
-#  Based on methods described in:
-#     Tuerlinckx, F. (2004). The efficient computation of the
-#     cumulative distribution and probability density functions
-#     in the diffusion model, Behavior Research Methods,
-#     Instruments, & Computers, 36 (4), 702-716.
-
-# """
+    Fnew = Fnew > Δ ? Fnew : 0
+
+    return    Fnew
+
+end
+
+function _add_outlier_cdf(y::Real, x::Real, p_outlier::Real; w_outlier::Real = 0.1)
+    #Ratcliff and Tuerlinckx, 2002 containment process
+    return y * (1 - p_outlier) + (x + (1. / (2 * w_outlier))) * w_outlier * p_outlier
+end
+
+function _p_outlier_in_range(p_outlier)
+    return (p_outlier >= 0) & (p_outlier <= 1)
+end
+
+"""
+    rand(dist::RatcliffDDM)
+
+Generate a random choice and rt for the Ratcliff Diffusion Model
+
+# Arguments
+- `dist`: model object for Ratcliff Diffusion Model. 
+- `method`: method simulating the diffusion process. 
+    "rejection" uses Tuerlinckx et al., 2001 rejection-based method for the general wiener process
+    "stochastic" uses the stochastic Euler method to directly simulate the stochastic differential equation
+
+# References
+
+    Tuerlinckx, F., Maris, E., Ratcliff, R., & De Boeck, P. (2001). 
+    A comparison of four methods for simulating the diffusion process. 
+    Behavior Research Methods, Instruments, & Computers, 33, 443-456.
+
+    Converted from Rhddmjagsutils.R R script by Kianté Fernandez
+    See also https://github.com/kiante-fernandez/Rhddmjags.
+"""
+function rand(rng::AbstractRNG, d::RatcliffDDM)
+    # method::Char = "rejection"
+    return _rand_rejection(rng, d)
+    # method::Char = "stochastic"
```

Comment:
Once you are confident you do not need commented code, can you delete throughout?